File: Modules/Default/Movie/MovieColumns.cs
using Serenity.ComponentModel;
using SerenityProjem.Default.Pages;
using System;
using System.ComponentModel;

namespace SerenityProjem.Default.Columns;

[ColumnsScript("Default.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MovieId { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }
    [DisplayName("Runtime in Minutes"), Width(150), AlignRight]
    public int Runtime { get; set; }
   
    [QuickFilter]
    public MovieKind Kind { get; set; }
    [Width(200),GenreListFormatter, QuickFilter ]
    public List<int> GenreList { get; set; }
}

File: Modules/ServerTypes/Default/MovieColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GenreListFormatter } from "../../Default/Movie/GenreListFormatter";
import { MovieRow } from "./MovieRow";
import { MovieKind } from "./Pages.MovieKind";

export interface MovieColumns {
    MovieId: Column<MovieRow>;
    Title: Column<MovieRow>;
    Description: Column<MovieRow>;
    Storyline: Column<MovieRow>;
    Year: Column<MovieRow>;
    ReleaseDate: Column<MovieRow>;
    Runtime: Column<MovieRow>;
    Kind: Column<MovieRow>;
    GenreList: Column<MovieRow>;
}

export class MovieColumns extends ColumnsBase<MovieRow> {
    static readonly columnsKey = 'Default.Movie';
    static readonly Fields = fieldsProxy<MovieColumns>();
}

[MovieKind, GenreListFormatter]; // referenced types